plasma_framework/contracts/src/exits/payment/PaymentInFlightExitable.sol

```diff
@@ -0,0 +1,300 @@
+        for (uint i = 0; i < exitData.inputTxs.length; i++) {
+            uint16 outputIndex = exitData.inputUtxosPos[i].outputIndex();
+            bytes32 outputGuard = exitData.inputTxs[i].outputs[outputIndex].outputGuard;
+
+            IPaymentSpendingCondition condition = PaymentSpendingConditionRegistry.spendingConditions(
+                exitData.inputUtxosTypes[i], exitData.inFlightTx.txType);
+            require(address(condition) != address(0), "Spending condition contract not found");
+
+            bool isSpentByInFlightTx = condition.verify(
+                outputGuard,
+                exitData.inputUtxosPos[i].value,
+                exitData.outputIds[i],
+                exitData.inFlightTxRaw,
+                uint8(i),
+                exitData.inFlightTxSigs[i]
+            );
+            require(isSpentByInFlightTx, "Spending condition failed");
+        }
+    }
+
+    function verifyInFlightTransactionDoesNotOverspend(StartExitData memory exitData) private pure {
+        PaymentTransactionModel.Transaction memory inFlightTx = exitData.inFlightTx;
+        for (uint i = 0; i < inFlightTx.outputs.length; i++) {
+            address token = inFlightTx.outputs[i].token;
+            uint256 tokenAmountOut = getTokenAmountOut(inFlightTx, token);
+            uint256 tokenAmountIn = getTokenAmountIn(exitData.inputTxs, exitData.inputUtxosPos, token);
+            require(tokenAmountOut <= tokenAmountIn, "Invalid transaction, spends more than provided in inputs");
+        }
+    }
+
+    function getTokenAmountOut(PaymentTransactionModel.Transaction memory inFlightTx, address token) private pure returns (uint256) {
+        uint256 amountOut = 0;
+        for (uint i = 0; i < inFlightTx.outputs.length; i++) {
+            if (inFlightTx.outputs[i].token == token) {
+                amountOut += inFlightTx.outputs[i].amount;
+            }
+        }
+        return amountOut;
+    }
+
+    function getTokenAmountIn(
+        PaymentTransactionModel.Transaction[] memory inputTxs,
+        UtxoPosLib.UtxoPos[] memory inputUtxosPos,
+        address token
+    )
+        private
+        pure
+        returns (uint256)
+    {
+        uint256 amountIn = 0;
+        for (uint i = 0; i < inputTxs.length; i++) {
+            uint16 oindex = inputUtxosPos[i].outputIndex();
+            PaymentOutputModel.Output memory output = inputTxs[i].outputs[oindex];
+            if (output.token == token) {
+                amountIn += output.amount;
+            }
+        }
+        return amountIn;
+    }
+
+    function startExit(StartExitData memory startExitData) private {
+        PaymentExitDataModel.InFlightExit storage ife = inFlightExits[startExitData.exitId];
+        ife.bondOwner = msg.sender;
+        ife.position = getYoungestInputUtxoPosition(startExitData.inputUtxosPos);
+        ife.exitStartTimestamp = block.timestamp;
+        setInFlightExitInputs(ife, startExitData.inputTxs, startExitData.inputUtxosPos);
```

Comment:
severe: don't we need to `setInflightExitOutputs` too?
@pgebal

Reply:
from slack 

> in the previous design we were setting outputs on piggyback

Reply:
they will be sent during piggybacking